libusb/core.c

```diff
@@ -2342,13 +2343,15 @@ int API_EXPORTED libusb_init(libusb_context **ctx)
 	list_add(&_ctx->list, &active_contexts_list);
 	usbi_mutex_static_unlock(&active_contexts_lock);
 
+	usbi_hotplug_init(_ctx);
+
 	if (usbi_backend.init) {
 		r = usbi_backend.init(_ctx);
 		if (r)
+			usbi_mutex_static_unlock(&default_context_lock);
```

Comment:
this doesn't look right: `err_io_exit` does unlock `default_context_lock` a few lines below